broker/lib/bosh/BoshOperationCache.js

```diff
@@ -0,0 +1,243 @@
+    });
+    return wrapper;
+  }
+
+  /**
+   * Fetches a limited number of BOSH deployment names from the Etcd Store
+   * 
+   * @param {Number of entries to fetch} numEntries 
+   */
+  getNEntries(numEntries) {
+    logger.debug(`Getting the first ${numEntries} entries in cache`);
+    const wrapper = new Promise((resolve, reject) => {
+      etcdConnector().getAll().prefix(CACHE_KEY).sort("Create", "None").limit(numEntries).keys().then(out => {
+        if (Array.isArray(out)) {
+          out = out.map(v => v.substring(CACHE_KEY.length));
+        } else {
+          throw new Error("Unexpected output");
+        }
+        resolve(out);
+      }).catch(err => reject(err));
+    });
+    return wrapper;
+  }
+
+  /**
+   * Fetches all BOSH deployment names in the Etcd store currently
+   */
+  getDeploymentNames() {
+    logger.debug('Getting the current deployment cache...');
+    const wrapper = new Promise((resolve, reject) => {
+      etcdConnector().getAll().prefix(CACHE_KEY).sort('Create', 'None').keys().then(out => {
+        if (Array.isArray(out)) {
+          out = out.map(v => v.substring(CACHE_KEY.length));
+        } else {
+          throw new Error("Unexpected output");
+        }
+        resolve(out);
+      }).catch(err => reject(err));
+    });
+    return wrapper;
+  }
+
+  /**
+   * Fetches BOSH task ID for a service instance from the Etcd store
+   * 
+   * @param {Service instance ID} serviceInstanceId 
+   */
+  getBoshTask(serviceInstanceId) {
+    logger.debug(`Getting bosh task ID for service instance id ${serviceInstanceId}`);
+    const wrapper = new Promise((resolve, reject) => {
+      const key = getTaskKey(serviceInstanceId);
+      etcdConnector().get(key).string().then(obj => {
+        resolve(obj);
+      }).catch(err => reject(err));
+    });
+    return wrapper;
+  }
+
+  /**
+   * Fetches parameters for a BOSH deployment operation from the Etcd store based on the name
+   * 
+   * @param {Deployment Name} name 
+   */
+  getDeploymentByName(name) {
+    logger.debug('Getting deployment for ', name);
+    const wrapper = new Promise((resolve, reject) => {
+      etcdConnector().get(getKey(name)).json().then(out => resolve(out)).catch(err => reject(err));
+    });
+    return wrapper;
+  }
+
+  /**
+   * Deletes a single BOSH deployment from the Etcd Store
+   * 
+   * @param {Deployment Name} deploymentName 
+   */
+  deleteDeploymentFromCache(deploymentName) {
+    logger.info('Removing deployment from cache', deploymentName);
+    const wrapper = new Promise((resolve, reject) => {
+      etcdConnector().delete().key(getKey(deploymentName)).then(out => resolve(out)).catch(err => reject(err));
+    });
+    return wrapper;
+  }
+
+  /**
+   * Deletes multiple BOSH deployment names from the Etcd store
+   * 
+   * @param {Deployment Names} deployments 
+   */
+  deleteDeploymentsFromCache(...deployments) {
+    const deploymentIdentifiers = _.flattenDeep(deployments);
```

Comment:
Why would a case arise where we may need to delete multiple deployments from cache?

Reply:
This was put in for the future monitoring requirements.